tests/store/tracking/test_sqlalchemy_store.py

```diff
@@ -534,10 +531,17 @@ def test_param_model(self):
             self.assertEqual(added_param.key, new_param.key)
 
     def test_run_needs_uuid(self):
+        mssql_regex = "Cannot insert the value NULL into column .+, table .+"
+        mysql_regex = "Field .+ doesn't have a default value"
+        pgsql_regex = "null value in column .+ of relation .+ violates not-null constrain"
+        sqlite_regex = "NOT NULL constraint failed"
+
         # Depending on the implementation, a NULL identity key may result in different
         # exceptions, including IntegrityError (sqlite) and FlushError (MysQL).
         # Therefore, we check for the more generic 'SQLAlchemyError'
-        with self.assertRaises(MlflowException) as exception_context:
+        with self.assertRaisesRegex(
+            MlflowException, rf"({mssql_regex}|{mysql_regex}|{pgsql_regex}|{sqlite_regex})"
+        ) as exception_context:
```

Comment:
Can we fix this block as follows?

```python
regex = {
    SQLITE: ...,
    POSTGRES: ...,
    MYSQL: ...,
    MSSQL: ...,
}[self.store._get_dialect()]
with self.assertRaisesRegex(MlflowException, regex)
```

Reply:
Sure thing! Yes, that would be much much better!